manifests/resource/location.pp

```diff
@@ -297,15 +312,19 @@
   Optional[String] $auth_basic_user_file                           = undef,
   Optional[String] $auth_request                                   = undef,
   Array $rewrite_rules                                             = [],
-  Integer[401,599] $priority                                       = 500,
+  Integer[401, 599] $priority                                      = 500,
   Boolean $mp4                                                     = false,
   Boolean $flv                                                     = false,
   Optional[String] $expires                                        = undef,
   Hash $add_header                                                 = {},
   Optional[Enum['on', 'off', 'always']] $gzip_static               = undef,
   Optional[Enum['on', 'off']] $reset_timedout_connection           = undef,
+  Optional[Variant[Array, String]] $access_log                     = undef,
+  Optional[Variant[Array, String]] $error_log                      = undef,
+  Optional[String] $format_log                                     = $nginx::http_format_log,
+  Optional[Enum['on', 'off']] $log_not_found                       = undef,
```

Comment:
A lot of data types are currently missing/inconsistent in the module, let's make the new ones perfect :wink: I think all these string must be non-empty, and arrays should only contain non-empty strings?

```suggestion
  Optional[Variant[Array[String[1]], String[1]]] $access_log       = undef,
  Optional[Variant[Array[String[1]], String[1]]] $error_log        = undef,
  Optional[String[1]] $format_log                                  = $nginx::http_format_log,
  Optional[Enum['on', 'off']] $log_not_found                       = undef,
```

Reply:
I get the idea and I think it is definitely a good one ;) 
However I just wonder if for syntax consistency sake doing it another PR that would process all data type in the module in one pass wouldn't be a wiser move wouldn't it ?
As you wish :)

Reply:
I would suggest you just adjust this for now.  You can adjust the code on your machine, `git commit --amend` to integrate these changes in your commit, and `git push -f` to update the PR.

If beyond these changes you want to help improving data types in the rest of the code, you are very welcome to  do so in another Pull Requests :smiley:.  Fixing everything would be a pain, and is likely to get in the way of other contributors work in progress, so I would suggest multiple successive PRs which focus on some aspects rather than a huge one with tries to fix everything.

Reply:
done ;)